src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/Recordable.kt

```diff
@@ -0,0 +1,54 @@
+package org.wycliffeassociates.otter.common.domain.content
+
+import org.wycliffeassociates.otter.common.data.model.ContentType
+import org.wycliffeassociates.otter.common.data.workbook.*
+
+class Recordable private constructor(
+    val textItem: TextItem? = null,
+    val audio: AssociatedAudio,
+    val start: Int? = null,
+    val end: Int? = null,
+    val sort: Int? = null,
+    val contentType: ContentType? = null
+) {
+    companion object {
+        fun build(bookElement: BookElement, component: Resource.Component): Recordable {
+            return when (bookElement) {
+                is Chapter -> Recordable(
+                    textItem = component.textItem,
+                    audio = component.audio,
+                    sort = component.sort,
+                    contentType = component.contentType
+                )
+                is Chunk -> Recordable(
+                    textItem = component.textItem,
+                    audio = component.audio,
+                    start = bookElement.start,
+                    end = bookElement.end,
+                    sort = component.sort,
+                    contentType = component.contentType
+                )
+                else -> throw IllegalStateException("Unsupported book element type found in Recordable.build")
+            }
+        }
+
+        fun build(chunk: Chunk) = Recordable(
+            textItem = chunk.text,
+            audio = chunk.audio,
+            start = chunk.start,
+            end = chunk.end
+        )
```

Comment:
We may need to change this function or add another build function later for when a user is recording an entire chapter. I'm not sure how this will work.